semantics/Makefile

```diff
@@ -87,36 +107,45 @@ clean:
 					 */c-nd-kompiled \
 					 */c-nd-thread-kompiled \
 					 .kompile-* \
-					 .depend-* \
-					 .build/*
+					 $(OUTPUT_DIR)
+
+GENERATED_MAKEFILE_DIR := $(OUTPUT_DIR)/make.d
 
-BASIC_DEPENDS := $(addprefix .depend-$(PROFILE)-basic-,c-translation cpp-translation c-cpp-linking c-cpp)
-SPECIAL_DEPENDS := .depend-$(PROFILE)-nd .depend-$(PROFILE)-nd-thread
-DEPENDS := $(BASIC_DEPENDS) $(SPECIAL_DEPENDS)
+$(GENERATED_MAKEFILE_DIR): ;@mkdir -p $@
 
-.PHONY: $(DEPENDS)
+BASIC_DEPENDS := $(addsuffix .dep.mk, \
+                   $(addprefix $(GENERATED_MAKEFILE_DIR)/basic-, \
+                      c-translation cpp-translation c-cpp-linking c-cpp))
+SPECIAL_DEPENDS := $(addsuffix .dep.mk, \
+                     $(addprefix $(GENERATED_MAKEFILE_DIR)/, \
+                        nd nd-thread))
 
 # See https://stackoverflow.com/a/4270649/6209703
 ifneq ($(MAKECMDGOALS),clean)
+  DEPENDS := $(BASIC_DEPENDS) $(SPECIAL_DEPENDS)
   -include $(DEPENDS)
 endif
 
-KDEP_FLAGS := $(KDEP_DEFAULT_FlAGS)
-KDEP_FLAGS += -I $(PROFILE_DIR)/semantics
-KDEP_FLAGS += -I $(PROFILE_DIR)/semantics/c
-KDEP_FLAGS += -I $(PROFILE_DIR)/semantics/cpp
-
-$(BASIC_DEPENDS): .depend-$(PROFILE)-basic-%:
-	$(eval TMP := $(shell mktemp))
-	$(KDEP) $(KDEP_FLAGS) -d ".build/$(PROFILE)/$*-kompiled" -- $*.k > $(TMP)
-	mv $(TMP) $@
-
-.depend-$(PROFILE)-nd:
-	$(eval TMP := $(shell mktemp))
-	$(KDEP) $(KDEP_FLAGS) -d ".build/$(PROFILE)/c-nd-kompiled" -- c-cpp.k > $(TMP)
-	mv $(TMP) $@
```

Comment:
I'm pretty sure things were done this way (with the temp files) to avoid some parallel make race. If you're confident this isn't necessary, I'm fine with removing them, though.

Reply:
I think it's taken care of by the fact that these are now dependencies of the `Makefile` itself. I haven't seen any failures with this -- locally or on `Jenkins`.